server/auth/sso/sso_test.go

```diff
@@ -153,3 +158,68 @@ func TestGetSessionExpiry(t *testing.T) {
 	}
 	assert.Equal(t, config.GetSessionExpiry(), 5*time.Hour)
 }
+
+func TestLoadSsoClientIdAndSecretFromEnv(t *testing.T) {
+	fakeClient := fake.NewSimpleClientset(ssoConfigSecret).CoreV1().Secrets(testNamespace)
+
+	os.Setenv("ARGO_CLIENT_ID", "sso-client-id-value")
+	os.Setenv("ARGO_CLIENT_SECRET", "sso-client-secret-value")
+
+	config := Config{
+		Issuer:               "https://test-issuer",
+		IssuerAlias:          "",
+		ClientIDEnvName:      "ARGO_CLIENT_ID",
+		ClientSecretEnvName:  "ARGO_CLIENT_SECRET",
+		RedirectURL:          "https://dummy",
+		CustomGroupClaimName: "argo_groups",
+	}
+	ssoInterface, err := newSso(fakeOidcFactory, config, fakeClient, "/", false)
+	assert.NoError(t, err)
+	ssoObject := ssoInterface.(*sso)
+	assert.Equal(t, "sso-client-id-value", ssoObject.config.ClientID)
+	assert.Equal(t, "sso-client-secret-value", ssoObject.config.ClientSecret)
+	assert.Equal(t, "argo_groups", ssoObject.customClaimName)
+	assert.Equal(t, "", config.IssuerAlias)
+	assert.Equal(t, 10*time.Hour, ssoObject.expiry)
+}
+
+func TestLoadSsoPrivateKeyFromEnv(t *testing.T) {
+	fakeClient := fake.NewSimpleClientset(ssoConfigSecret).CoreV1().Secrets(testNamespace)
+
+	generatedKey, err := rsa.GenerateKey(rand.Reader, 2048)
+	assert.NoError(t, err)
+	assert.NotEmpty(t, generatedKey)
+
+	privateKeyBlob := x509.MarshalPKCS1PrivateKey(generatedKey)
+	assert.NotEmpty(t, privateKeyBlob)
+
+	privateKeyBlobEncoded := base64.StdEncoding.EncodeToString(privateKeyBlob)
+	assert.NotEmpty(t, privateKeyBlobEncoded)
+
+	os.Setenv("ARGO_PRIVATE_KEY", privateKeyBlobEncoded)
+
+	ctx := context.Background()
+	_, err = fakeClient.Create(ctx, &apiv1.Secret{
+		ObjectMeta: metav1.ObjectMeta{Name: secretName},
+		Data:       map[string][]byte{},
+	}, metav1.CreateOptions{})
+	assert.NoError(t, err)
```

Comment:
hmm, creating an empty PKCS secret that [would fail normally](https://github.com/argoproj/argo-workflows/blob/375a860b51e22378ca529da77fe3ed1ecb8e6de6/server/auth/sso/sso_test.go#L146) relies on a good bit of indirection. that's confusing and is also not specified in a comment here either.

instead, we should be direct that no PKCS secret is created by asserting that a `get` finds no secret at the end